domain/goal/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/GoalEventService.kt

```diff
@@ -10,10 +10,10 @@ interface GoalEventService {
   /**
    * Implementations providing custom code for when a [Goal] is created.
    */
-  fun goalCreated(createdGoal: Goal)
+  fun goalCreated(prisonNumber: String, createdGoal: Goal)
 
   /**
    * Implementations providing custom code for when a [Goal] is updated.
    */
-  fun goalUpdated(updatedGoal: Goal)
+  fun goalUpdated(prisonNumber: String, updatedGoal: Goal, existingGoal: Goal)
```

Comment:
Really really minor; and dont feel you have to do this, but I wonder what you think about changing the order of args and the name of the "existing" goal - something like this perhaps?
```
fun goalUpdated(prisonNumber: String, previousGoal: Goal, updatedGoal: Goal)
```
or
```
fun goalUpdated(prisonNumber: String, goalPreUpdate: Goal, goalPostUpdate: Goal)
```
or something like that ... you get the gist ...

Reply:
It's very important, not minor! 😉 

Funnily enough, I looked at the order of the arguments last night and thought the existing/previous goal should come first.

I'm not sure about `previous`, because it's the same goal, but I'll have a think about the name 👍

Reply:
Actually, I've gone with previous, since it was the previous state of the goal at this point in time!